server/handler.go

```diff
@@ -103,6 +106,41 @@ func (h *Handler) ComQuery(
 
 	var r *sqltypes.Result
 	var proccesedAtLeastOneBatch bool
+
+	rowchan := make(chan sql.Row)
+	errchan := make(chan error)
+	quit := make(chan struct{})
```

Comment:
[optional] You could put `defer close(quit)` here, instead of calling it in multiple places below. As far as I can see, anytime this function returns, you'd want `quit` closed anyway.

Reply:
I prefer to make it explicit so the code reader can see where we are actively signaling the goroutine.